src/vgithub/vgithub.ml

```diff
@@ -781,30 +788,36 @@ module Sync (API: API) (DK: Datakit_S.CLIENT) = struct
     Log.debug (fun l ->
         l "sync pub:%s priv:%s" (DK.Branch.name pub) (DK.Branch.name priv)
       );
-    let event_diff last_t current =
-      of_commit current >>*= fun current_t ->
-      let diff = UserRepoSet.sdiff last_t.user_repos current_t.user_repos in
-      Log.debug (fun l -> l "user-repo-diff: %a" UserRepoSet.pp diff);
+    let user_repo_diff old_t c =
+      of_commit c >>*= fun current_t ->
+      let diff = XRepoSet.sdiff old_t.repos current_t.repos in
+      Log.debug (fun l -> l "user-repo-diff: %a" XRepoSet.pp diff);
       ok diff
     in
-    let github_calls last_t current =
+    let call_github_api old_t current =
       of_commit current >>*= fun current_t ->
-      call_github_api ~token ~old:last_t current_t >>*= fun () ->
+      call_github_api ~token ~old:old_t current_t >>*= fun () ->
       ok ()
     in
     let prune current =
       of_commit current >>*= fun last_t ->
       prune last_t priv >>*= fun () ->
       ok ()
     in
-    let merge c =
+    let merge old_t c =
       DK.Branch.with_transaction pub (fun tr ->
           DK.Transaction.merge tr c >>*= fun (_, conflicts) ->
           if conflicts <> [] then failwith "TODO";
           let msg = Fmt.strf "Merging with %s" @@ DK.Branch.name priv in
-          DK.Transaction.commit tr ~message:msg
-        ) >>*= fun _ ->
-      ok ()
+          of_commit c >>*= fun priv ->
+          of_tree (E ((module DK.Transaction), tr)) >>*= fun pub ->
+          (if compare_snapshot old_t.priv priv = 0 then
```

Comment:
@talex5 I think this does the right thing: e.g. it merges only if the private branch has changed.